src/main/java/com/sforce/cd/apexUnit/client/connection/ConnectionHandler.java

```diff
@@ -106,21 +115,16 @@ public static ConnectionHandler getConnectionHandlerInstance() {
 			
 	private PartnerConnection createConnection() {
 		if (connection == null) {
-			ConnectorConfig config = new ConnectorConfig();
-			config.setUsername(CommandLineArguments.getUsername());
-			config.setPassword(CommandLineArguments.getPassword());
+			ConnectorConfig config = createConfig();
 			config.setAuthEndpoint(CommandLineArguments.getOrgUrl() + "/services/Soap/u/" + SUPPORTED_VERSION);
-			
-			if (CommandLineArguments.getProxyHost() != null &&  CommandLineArguments.getProxyPort() !=null ){
-				LOG.debug("Setting proxy configuraiton to " + 
-					CommandLineArguments.getProxyHost() + 
-					" on port " + CommandLineArguments.getProxyPort() );
-				config.setProxy( CommandLineArguments.getProxyHost() , CommandLineArguments.getProxyPort());
-			}	
-			
+			config.setSessionRenewer(new SFDCSessionRenewer());
+			LOG.info("Default connection time out value is: " + config.getConnectionTimeout());
+			config.setConnectionTimeout(MAX_TIME_OUT_IN_MS_INT);
```

Comment:
Added checks for -ve and null values.

For my test org, the value was set to zero. Not sure if this will be the case across all orgs.. We decided to customize this because we didn't want to wait for connection exception to be thrown in order to trigger the session renewer. We set this value so that we can detect the issues with connection quickly and get session renewer into action.